### PR TITLE
feat: Llama3

### PR DESCRIPTION
**Pull request by kir-gadjello**
I added support for Llama3 family of LLMs, I tested it with Llama3-8B and Llama3-70B and found it to perform well. I also touched up the docs.

There are three new formats for interacting with llms: llama3, llama3-grammar, llama3-hints-grammar.

### CODE REVIEW

File: memgpt/local_llm/llm_chat_completion_wrappers/llama3.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+# Initial information about them is provided in the core memory user block
+# Make sure to introduce yourself to them
+# Your inner thoughts should be private, interesting, and creative
+# Do NOT use inner thoughts to communicate with the user
+# Use send_message to communicate with the user"""
+# Don't forget to use send_message, otherwise the user won't see your message"""
+
+
+class LLaMA3InnerMonologueWrapper(LLMChatCompletionWrapper):
+    """ChatML-style prompt formatter, tested for use with https://huggingface.co/ehartford/dolphin-2.5-mixtral-8x7b#training"""
```
Comment (znfgnu) [inline memgpt/local_llm/llm_chat_completion_wrappers/llama3.py:31]: Shouldn't link to the website be updated?

File: memgpt/local_llm/llm_chat_completion_wrappers/llama3.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+# If you receive new important information about the user (or yourself), you immediately update your memory with core_memory_replace, core_memory_append, or archival_memory_insert"""
+
+FIRST_PREFIX_HINT = """# Reminders:
+# This is your first interaction with the user!
+# Initial information about them is provided in the core memory user block
+# Make sure to introduce yourself to them
+# Your inner thoughts should be private, interesting, and creative
+# Do NOT use inner thoughts to communicate with the user
+# Use send_message to communicate with the user"""
+# Don't forget to use send_message, otherwise the user won't see your message"""
```
[the remark this answers is not in the document]
Comment (kir-gadjello) [reply memgpt/local_llm/llm_chat_completion_wrappers/llama3.py:27]: This file is an almost verbatim copy of the pre-existing source file https://github.com/cpacker/MemGPT/blob/main/memgpt/local_llm/llm_chat_completion_wrappers/chatml.py containing the same code.

I agree that the comment is interesting and might reflect some mistake by original author, but I think the system prompt is very sensitive part of memgpt and should not be fixed lightly, you need to do it in a separate PR.

This PR should be about basic baseline support of Llama3 in line with other generic local LLMs, I think.

[the remark this answers is not in the document]
Comment (kir-gadjello) [reply memgpt/local_llm/llm_chat_completion_wrappers/llama3.py:27]: (Also, I think those system prompt hint suffixes should be shared, IMO, and @cpacker might want to refactor them in the future).

**Comment by kir-gadjello** [comment]
Are we ready to merge?

**Review by cpacker — APPROVED** [review]
Awesome PR, thank you! Sorry for the delay in the review

**Pull request merged**